exporter/exporterhelper/metrics_batch.go

```diff
@@ -8,202 +8,319 @@ import (
 	"errors"
 
 	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+	"go.opentelemetry.io/collector/exporter/exporterhelper/internal/sizer"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 )
 
 // MergeSplit splits and/or merges the provided metrics request and the current request into one or more requests
 // conforming with the MaxSizeConfig.
 func (req *metricsRequest) MergeSplit(_ context.Context, cfg exporterbatcher.SizeConfig, r2 Request) ([]Request, error) {
+	var sz sizer.MetricsSizer
+	switch cfg.Sizer {
+	case exporterbatcher.SizerTypeItems:
+		sz = &sizer.MetricsCountSizer{}
+	case exporterbatcher.SizerTypeBytes:
+		sz = &sizer.MetricsBytesSizer{}
+	default:
+		return nil, errors.New("unknown sizer type")
```

Comment:
Add tests for this. Maybe also create a separate PR to add a test for logs as well.